src/theory/quantifiers/ceg_instantiator.h

```diff
@@ -41,14 +41,24 @@ class CegqiOutput {
 class Instantiator;
 
 
+class TermProperties {
+public:
+  TermProperties() : d_type(-1) {}
+  int d_type;
+  // for arithmetic
+  Node d_coeff;
+};
+
 //solved form, involves substitution with coefficients
 class SolvedForm {
 public:
   std::vector< Node > d_vars;
   std::vector< Node > d_subs;
+  //
   std::vector< Node > d_coeff;
   std::vector< int > d_btyp;
-  std::vector< Node > d_has_coeff;
+  //std::vector< TermProperties > d_props;
+  std::vector< Node > d_has_coeff;//d_has_prop;
```

Comment:
What is the meaning of `//d_has_prop;`?

Reply:
While preparing this commit, I realized several things should be refactored, including generalizing "has coefficient" to "has properties" where coefficients are LIA-specific but this infrastructure needs to be generic for theories.
Most of these comments are reminders for the next commit I will make. In the future I will make this clearer.